### ISSUE TITLE
Showing extra day on all-day ICS event

### ISSUE THREAD

**Issue by kkoyung** [anufrievroman/calcure#89]
### Issue

Calcure displays all-day ICS event with one extra day.

Here is an example of ICS event.
```ics
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//PIMUTILS.ORG//NONSGML khal / icalendar //EN
BEGIN:VEVENT
SUMMARY:two days event
DTSTART;VALUE=DATE:20231203
DTEND;VALUE=DATE:20231205
DTSTAMP:20231202T163238Z
UID:LSFHOYXWW9PRMRM1UMD0FUFRFSPJYA34JSOG
SEQUENCE:0
CATEGORIES:
END:VEVENT
END:VCALENDAR
```

From [Section 3.6.1](https://datatracker.ietf.org/doc/html/rfc5545#section-3.6.1) of RFC 5545 (the spec of iCalendar), the start date `DTSTART` is inclusive, and the end date `DTEND` is non-inclusive. So, this example is a two days event (Dec 3 and 4). However, calcure shows it as a three days event (Dec 3, 4 and 5). One more extra day.

![screenshot_2023-12-02_19:09:25](https://github.com/anufrievroman/calcure/assets/11942650/f6b1ec85-b9d5-4cab-9bc0-b2c7c264991d)

### Inspection

I think the problem comes from calculating the repetition in `loaders.py`.
https://github.com/anufrievroman/calcure/blob/f363ccc68abbe1921ccf47e15941b93bed62defe/calcure/loaders.py#L430
Since the end data is non-inclusive, we don't need to add 1 to the difference in days.

**Comment by anufrievroman** [comment]
I think I fixed it in e3787ed3cdb8ce558f4bc1b5dcd157f79d808a1d Please try the main branch

**Comment by kkoyung** [comment]
The all day events are shown at correct dates now. It works perfectly.
